lms/services/blackboard_api/client.py

```diff
@@ -113,4 +114,35 @@ def group_set_groups(self, course_id, group_set_id):
             "GET",
             f"/learn/api/public/v2/courses/uuid:{course_id}/groups/sets/{group_set_id}/groups",
         )
-        return BlackboardListGroupSetGroups(response).parse()
+        return BlackboardListGroupsSchema(response).parse()
+
+    def course_groups(self, course_id, group_category_id=None, user_id=None):
+        response = self._api.request(
+            "GET",
+            f"/learn/api/public/v2/courses/uuid:{course_id}/groups",
+        )
+        groups = BlackboardListGroupsSchema(response).parse()
+
+        if group_category_id:
+            groups = [
+                group
+                for group in self.course_groups(course_id)
+                if group["groupSetId"] == group_category_id
+            ]
+
+        if user_id:
+            groups_by_id = {group["id"]: group for group in groups}
```

Comment:
Create a dict of the groups by ID to retrieve them later as the responses will be out of order.